### DISCUSSION TITLE
I was wondering how has StreetComplete managed to render the map so well, on Android.

### DISCUSSION THREAD

**Discussion by xidsyed** [streetcomplete/StreetComplete#4665] [Q&A] [answered]
I am a beginner android developer and was looking to create a map like interface in an app that I want to work on. I was wondering what library does StreetComplete use to render its map? I tried to go digging but only got more confused. can someone point me in the right direction?

Also how is it so performant? Google maps is super laggy, apple maps are comparable, is Apple maps also based on OSM, I think i read that somewhere.

**Comment by matkoniecz** [comment, marked as answer]
StreetComplete is using Mapzen Tangram ( https://www.mapzen.com/documentation/ ) but I would not recommend it as project died.

Maplibre ( https://docs.maptiler.com/maplibre-gl-native-android/ ) is better for new projects.

**Comment by xidsyed** [comment]
@matkoniecz are there any other good active open source projects with a good community such as this one that you are aware of that aren't based around croud sourcing maps... i would love to learn from contributing to this project, except i would never use it. so it would become a chore very quickly :(
